clustered-indexes.md

```diff
@@ -1,188 +1,225 @@
+TiDB specific comment syntax supports wrapping the keywords `CLUSTERED` and `NONCLUSTERED` in a comment. The result of `SHOW CREATE TABLE` also contains TiDB specific SQL comments. MySQL databases and TiDB databases of a lower version can identify and execute DDL statements with these comments.
+
+### Compatibility with TiDB ecosystem tools
+
+The clustered indexes feature is only compatible with the following ecosystem tools in v5.0 and later versions:
+
+- Backup and recovery tools: BR, Dumpling, and TiDB Lightning.
+- Data migration and replication tools: DM and TiCDC.
+
+However, you cannot convert a table with non-clustered indexes to a table with clustered indexes by using the v5.0 BR to backup and recover tables, and vice versa.
+
+### Compatibility with other TiDB features
+
+For a table with a combined primary key or a single non-integer primary key, if you change the primary key from a non-clustered index to a clustered index, the keys of its row data change as well. Therefore, `SPLIT TABLE BY/BETWEEN` statements that are executable on TiDB v5.0 and lower versions, are no longer workable in v5.0 and higher versions of TiDB. If you want to split a table with clustered indexes using `SPLIT TABLE BY/BETWEEN`, you need to provide the value of the primary key column, instead of specifying an integer value. See the following example:
 
 ```sql
-CREATE TABLE t1 (
- guid CHAR(32) NOT NULL PRIMARY KEY,
- b BIGINT,
- INDEX(b)
-);
+mysql> create table t (a int, b varchar(255), primary key(a, b) clustered);
+Query OK, 0 rows affected (0.01 sec)
+mysql> split table t between (0) and (1000000) regions 5;
+ERROR 1105 (HY000): Split table region lower value count should be 2
+mysql> split table t by (0), (50000), (100000);
+ERROR 1136 (21S01): Column count doesn't match value count at row 0
+mysql> split table t between (0, 'aaa') and (1000000, 'zzz') regions 5;
++--------------------+----------------------+
+| TOTAL_SPLIT_REGION | SCATTER_FINISH_RATIO |
++--------------------+----------------------+
+|                  4 |                    1 |
++--------------------+----------------------+
+1 row in set (0.00 sec)
+mysql> split table t by (0, ''), (50000, ''), (100000, '');
++--------------------+----------------------+
+| TOTAL_SPLIT_REGION | SCATTER_FINISH_RATIO |
++--------------------+----------------------+
+|                  3 |                    1 |
++--------------------+----------------------+
+1 row in set (0.01 sec)
 ```
 
-Because the pointer to the `guid` is a `char(32)`, each index value for `b` will now require approximately `8 + 32 = 40 bytes` (a `BIGINT` value requires 8 bytes for storage). This compares to `8 + 8 = 16 bytes` for non-clustered tables. The exact storage requirements will differ after compression has been applied.
+The attribute [`AUTO_RANDOM`](/auto-random.md) can only be used in tables with clustered indexes. Otherwise, TiDB returns the following error:
```

Comment:
```suggestion
The attribute [`AUTO_RANDOM`](/auto-random.md) can only be used on clustered indexes. Otherwise, TiDB returns the following error:
```